src/player/Player.cpp

```diff
@@ -98,25 +85,27 @@ void Player::handleKeyboardInput(const Uint8 *state) {
     }
 
     //Inventory inputs
-    if (state[SDL_SCANCODE_1]) {
+    if (state[SDL_SCANCODE_1]){
         marine->inventory.switchCurrent(0);
-    } else if (state[SDL_SCANCODE_2]) {
+    } else if (state[SDL_SCANCODE_2]){
         marine->inventory.switchCurrent(1);
-    } else if (state[SDL_SCANCODE_3]) {
+    } else if (state[SDL_SCANCODE_3]){
         marine->inventory.switchCurrent(2);
     }
 
     //Weapon input
-    if(state[SDL_SCANCODE_R]) {
+    if(state[SDL_SCANCODE_R]){
         marine->inventory.getCurrent()->reloadClip();
     }
-    if(state[SDL_SCANCODE_E]) {
+    if(state[SDL_SCANCODE_E]){
+        marine->checkForPickUp();
         const int currentTime = SDL_GetTicks();
 
         if(currentTime > (pickupTick + pickupDelay)) {
             pickupTick = currentTime;
             const int checkTurret = marine->checkForPickUp();
-            if (checkTurret > -1 && !holdingTurret) {
+            if (checkTurret > -1 && holdingTurret == false)
+            {
```

Comment:
You get that bracket back into line or I'll have it dragged out into the field and shot 🍆